graphql/src/main/java/com/linecorp/armeria/server/graphql/GraphqlServiceBuilder.java

```diff
@@ -258,11 +286,17 @@ public GraphqlService build() {
             configurer.configure(builder);
         }
 
-        final DataLoaderRegistry dataLoaderRegistry = new DataLoaderRegistry();
-        final List<Consumer<? super DataLoaderRegistry>> dataLoaderRegistries =
-                dataLoaderRegistryConsumers.build();
-        for (Consumer<? super DataLoaderRegistry> configurer : dataLoaderRegistries) {
-            configurer.accept(dataLoaderRegistry);
+        final Function<? super ServiceRequestContext, ? extends DataLoaderRegistry> dataLoaderRegistryFactory;
+        if (this.dataLoaderRegistryFactory != null) {
+            dataLoaderRegistryFactory = this.dataLoaderRegistryFactory;
+        } else if (dataLoaderRegistryConsumers != null) {
+            final DataLoaderRegistry dataLoaderRegistry = new DataLoaderRegistry();
+            for (Consumer<? super DataLoaderRegistry> configurer : dataLoaderRegistryConsumers.build()) {
+                configurer.accept(dataLoaderRegistry);
+            }
+            dataLoaderRegistryFactory = ctx -> dataLoaderRegistry;
+        } else {
+            dataLoaderRegistryFactory = ctx -> new DataLoaderRegistry();
```

Comment:
It is an unexpected state if both `dataLoaderRegistryFactory` and `dataLoaderRegistryConsumers` are not null. Should we raise an `IllegalStateException` for that?

Reply:
On the user side, I thought that the DataLoader would not be needed all the time. That's why I only put in an empty dataLoaderRegistry. WDYT? 🤔

Reply:
It is okay to set an empty `DataLoaderRegistry` when not explicitly set. 
I am worried a situation where `dataLoaderRegistryConsumers != null && dataLoaderRegistryFactory != null` is true. But it seems never to happen because the builder methods validate them.
Let's keep it as is and add an assertion for clarity.
```java
assert dataLoaderRegistryFactory == null && dataLoaderRegistryConsumers == null;
```

Reply:
I think so too. We'll have to remove the deprecated method someday. 🤣